.github/workflows/docs.yml

```diff
@@ -0,0 +1,51 @@
+name: docs
+on:
+  push:
+    branches: [main, feature/doxygen]
+  workflow_dispatch:
+  workflow_call:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  generate_docs:
+    name: Generate and upload docs
+    runs-on: ubuntu-20.04
+    container:
+      image: godexsoft/functional-ci:0.0.1
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+            
+      - name: Export docs
+        continue-on-error: false
+        shell: bash
+        id: run_cmake
+        run: |
+          mkdir .build
+          cd .build
+          cmake -DDOCS=True ..
+          cmake --build . --target export_docs
+      
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: ./.build/docs
```

Comment:
Ah right. One way to "improve" it a bit is to use `docs` as `docId` (e.g. https://liftwizard.io/docs/configuration/json5-configuration)

But also wonder (haven't tried) if you pass an empty string or maybe forward slash as docId. 

The reason for docId internal requirement is so urls can be treated the same way /docs/my-page and /docs/my-page/ (extra slash). Otherwise relative linking (..) would be messed up. 

If you go with `docs` then you can imagine having `blog` or something like that. Not ideal trade off I understand. 

Let me know if empty docId doesn't work and I will check the code to make it work.

Reply:
Thanks for the suggestions. Trying now.
- If i specify empty (``) or a space (` `) for doc-id, then i get `Missing argument for option: doc-id`
- Specifying a forward slash (`/`) yields an exception: 
  ```
  generating the rest of HTML pages
  deploying index.md to /index.html
  Exception in thread "main" java.lang.RuntimeException: error during rendering of index: java.nio.file.AccessDeniedException: /index.html
  ```
Looks like this use-case is just not supported atm.
Perhaps it will be easy to fix. I like the forward slash idea. I feel like it's almost working except for the permissions 🤞

Reply:
https://github.com/testingisdocumenting/znai/pull/1188

Reply:
Hi @MykolaGolubyev, hope you are doing well :+1: 
I see that the issue got merged some time ago - any release plans for the coming future? :bow:

Reply:
Thanks for the reminder! Released 1.68
Let me know how it goes. I can make another release soon

Reply:
@MykolaGolubyev thanks for your help on this! 

Znai now generates to root correctly when i specify `--doc-id '""'`.
Note that it does not work without the extra single quotes as it errors out the same as before with "empty" otherwise.

However, now it appears to ignore [my index page and opens straight to the first item in TOC](http://libfn.org). Any quick way to disable this behaviour you know of?
I have not yet checked the other features you added since 1.85. Will keep you posted when I do. 👍

Reply:
Could you please try to remove doc-id parameter altogether?

Reply:
ah sorry, there is an automatic name generation. I am looking into why index automatically redirects.
Feel free to provide more feedback so I can pack them all together for the next release

Reply:
> Could you please try to remove doc-id parameter altogether?

This does not appear to work. It creates the `no-id-specified` folder on export then.

> Feel free to provide more feedback so I can pack them all together for the next release

Sorry, did not find the time to write more documentation yet. So i don't really have feedback outside of what we covered before in the discussion on this pr. 🙇

Reply:
The fix was released as 1.69

Reply:
I have switched to the new version and it seems to do the right thing now 👍 🙇

Reply:
Let me know if you have docs deployed somewhere - I could take a look and maybe suggest a thing or two if you want. 
If you have some energy left, could you please post something here - https://github.com/testingisdocumenting/znai/discussions/824?sort=new

Reply:
@MykolaGolubyev latest docs are deployed to [libfn.org](https://libfn.org/) but we are yet to actually write the contents :rage4: